src/main/java/picocli/CommandTokenizer.java

```diff
@@ -0,0 +1,371 @@
+package picocli;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+import picocli.CommandLine;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.Parameters;
+
+public class CommandTokenizer {
+
+		private String[] tokens;
+		private int position = 0;
+		// Characters that cimment out until a newline
+		private List<Character> commentChars = new ArrayList<Character>(Arrays.asList('#'));
+		// Characters that escape other special characters
+		private List<Character> escapeChars = new ArrayList<Character>(Arrays.asList('\\'));
+		// Characters to be escaped automatically
+		private List<Character> escapedChars = new ArrayList<Character>();
+		// Characters that begin/end quotes
+		private List<Character> quoteChars = new ArrayList<Character>(Arrays.asList('"', '\''));
+		// Characters to be treated as whitespace
+		private List<Character> whitespaceChars = new ArrayList<Character>(Arrays.asList(' '));
+		// Characters to use to represent a new line
+		private List<String> eolPatterns = new ArrayList<String>(Arrays.asList("\r\n", "\r", "\n"));
+		// Remove blanks at start and end of a string
+		private boolean trimBlanks = true;
+
+		public CommandTokenizer(String cmd) {
```

Comment:
Both constructors start parsing the specified input immediately before users have had a chance to configure the tokenizer. It may be better to postpone the call to `parse` until the user calls the `tokens` or `nextToken` method.

Reply:
Oops, you're right. Didn't think about that, will fix.